.github/workflows/quality-assurance.yml

```diff
@@ -85,22 +86,22 @@ jobs:
         run: ATRIUM_ANDROID_JAR="$PWD/android-jar-cache/android.jar" ./gradlew checkDexer
 
   forwardCompatibility:
-    needs: dexer
+    needs: []  #Not sure if it should be left empty or if it should contain generate-matrix
     env:
       JAVA_VERSION: 17
       KOTLIN_VERSION: ${{ matrix.kotlin_version }}
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        kotlin_version: ['1.5', '1.6', '1.7', '1.8', '1.9', '2.0']
+        kotlin_version: [ '1.5', '1.6', '1.7', '1.8', '1.9', '2.0' ]
     steps:
       - uses: actions/checkout@v4
       - name: Set up JDK 17
         uses: actions/setup-java@v4
         with:
-          distribution: 'adopt'
-          java-version: 17
+          distribution: ${{ matrix.java_distribution }}
```

Comment:
keep adopt and 17 we define the matrix differently in this job (i.e. we only define `kotlin_version`, see further above)